addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3431,9 +3431,23 @@ export class OdooEditor extends EventTarget {
         if (!rangeContent.hasChildNodes()) {
             return;
         }
+        const closestLi = closestElement(selection.anchorNode, 'li');
+        const commonAncestorElement = closestElement(range.commonAncestorContainer);
+        if (commonAncestorElement && !isBlock(rangeContent.firstChild)) {
+            // Get the list of ancestor elements starting from the provided
+            // commonAncestorElement up to the closest li or block-level element.
+            const blockOrLi = closestLi || closestBlock(commonAncestorElement);
+            const ancestorsList = [commonAncestorElement, ...ancestors(commonAncestorElement, blockOrLi)];
+            // Wrap rangeContent with clones of their ancestors to keep the styles.
+            for (const ancestor of ancestorsList) {
+                const clone = ancestor.cloneNode();
+                clone.append(...rangeContent.childNodes);
+                rangeContent.appendChild(clone);
+            }
```

Comment:
Currently, we have inside `_onClipboardCopy` code blocks to adjust the range content when the selection involves lists or tables. We don't want change the `rangeContent` before such code blocks are run.

I suggest you move this code section to **after** such cases are handled (right before `const dataHtmlElement = document.createElement('data');`.

And, unless I am missing something (which is quite likely to be the case! :grimacing: ), you won't have to handle `li` at this point. So you can drop the `closestLi` in your code, getting the ancestors up to the closest block element. If the selected test is inside some tags nested in a `li` element, the `li` will be returned by `closestBlock(commonAncestorElement)`.

Reply:
@cammarosano I have updated the requested changes!

Reply:
Niiice! Thanks!

I have a tip when force-pushing changes requested by a reviewer: first push the changes you've just done, without the rebase. Then, in case you need to rebase (usually we do), rebase then push again. This way we have 2 "Compare" diffs,  with the first one containing purely the requested changes. :wink: 
Don't worry if you forget to do it, though, I often times forget it...